### PR TITLE
Libvirt-actuator bootstrap command

### PR DESCRIPTION
**Pull request by ingvagabund**
alt text

### CODE REVIEW

File: cmd/libvirt-actuator/main.go [hunk 1]
```diff
@@ -123,7 +144,244 @@ func init() {
...
+
+func bootstrapCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "bootstrap",
+		Short: "Bootstrap kubernetes cluster with kubeadm",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			libvirturi := cmd.Flag("libvirt-uri").Value.String()
+			if libvirturi == "" {
+				return fmt.Errorf("--libvirturi needs to be set")
```
Comment (spangenberg) [inline cmd/libvirt-actuator/main.go]:
```suggestion
				return fmt.Errorf("--libvirt-uri needs to be set")
```

File: cmd/libvirt-actuator/main.go [hunk 2]
```diff
@@ -123,7 +144,244 @@ func init() {
...
+			}
+
+			inclusterlibvirturi := cmd.Flag("in-cluster-libvirt-uri").Value.String()
+			if inclusterlibvirturi == "" {
+				return fmt.Errorf("--in-cluster-libvirturi needs to be set")
```
Comment (spangenberg) [inline cmd/libvirt-actuator/main.go]:
```suggestion
				return fmt.Errorf("--in-cluster-libvirt-uri needs to be set")
```

**Comment by spangenberg** [comment]
Apart from the two typos it looks good to me.

**Comment by ingvagabund** [comment]
@spangenberg comments addressed

@enxebre @bison @frobware PTAL, tested it locally, it deploys the cluster with CRD based machine API

File: cmd/libvirt-actuator/main.go [hunk 3]
```diff
@@ -109,7 +133,244 @@ func init() {
...
+
+func createSecretAndWait(f *framework.Framework, secret *apiv1.Secret) error {
+	_, err := f.KubeClient.CoreV1().Secrets(secret.Namespace).Create(secret)
+	if err != nil {
+		return err
+	}
+
+	err = wait.Poll(framework.PollInterval, framework.PoolTimeout, func() (bool, error) {
+		if _, err := f.KubeClient.CoreV1().Secrets(secret.Namespace).Get(secret.Name, metav1.GetOptions{}); err != nil {
+			return false, nil
```
Comment (frobware) [inline cmd/libvirt-actuator/main.go]: This, and the following `return false, nil`, can be collapsed to:

    return err != nil, err

File: cmd/libvirt-actuator/main.go [hunk 4]
```diff
@@ -109,7 +133,244 @@ func init() {
...
+			}
+
+			lcw, err := libvirtclient.NewLibvirtClient(libvirturi)
+			if err != nil {
+				return err
+			}
+
+			// Wait until the instance has the ip address
+			var masterMachinePrivateIP string
+			err = wait.Poll(pollInterval, timeoutPoolInterval, func() (bool, error) {
```
Comment (frobware) [inline cmd/libvirt-actuator/main.go]: We don't appear to check the state of this error value.

Comment (ingvagabund) [reply cmd/libvirt-actuator/main.go]: Done

**Comment by ingvagabund** [comment]
@frobware PTAL

**Comment by frobware** [comment]
> @frobware PTAL

/lgtm

**Pull request merged**